Fix EZP-23693: Http Headers are not set when using FastCGI

blankse opened the pull request:
JIRA: https://jira.ez.no/browse/EZP-23693

blankse commented:
ping @bdunogier @dpobel

andrerom commented:
@blankse as asked in the issue, any idea why this was not detected before? fast cgi has been used by some for years, and fpm for some years now as well. Is this for a specific setup or fpm version?

dpobel commented on lib/ezutils/classes/ezhttptool.php:
```diff
@@ -615,17 +609,31 @@ static function redirect( $path, $parameters = array(), $status = false, $encode
+    static function sendHTTPResponseCode( $httpResponseCode )
     {
-        header( $headerName .': '. $headerData );
+        eZHTTPTool::headerVariable('X-PHP-Response-Code', $httpResponseCode, true, $httpResponseCode );
```
nitpick CS: missing space after the opening parenthesis

dpobel commented:
+1 (besides the CS nitpick)

blankse commented:
@andrerom  I only have a problem with the login redirect which I solved in a other PR (https://github.com/ezsystems/ezpublish-kernel/pull/1118). I make the changes because the description in JIRA.

@dpobel I corrected the nitpick and rebased to master.

andrerom commented:
Ok, so if that is the case this is not relevant anymore then. Legacy is in "mature" mode, so we don't merge big changes unless we have to..

blankse commented:
OK, then I close it.

andrerom commented:
Thanks @blankse, and sorry for being such a strict maintainer ;)

blankse commented:
No problem. That sounds good :)
